packages/heartbeat/src/class-heartbeat.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * Jetpack Heartbeat package.
+ *
+ * @package  automattic/jetpack-heartbeat
+ */
+
+namespace Automattic\Jetpack;
+
+use Automattic\Jetpack\A8c_Mc_Stats;
+
+/**
+ * Heartbeat sends a batch of stats to wp.com once a day
+ */
+class Heartbeat {
+
+	/**
+	 * Holds the singleton instance of this class
+	 *
+	 * @since 2.3.3
+	 * @var Heartbeat
+	 */
+	private static $instance = false;
+
+	/**
+	 * Cronjob identifier
+	 *
+	 * @var string
+	 */
+	private $cron_name = 'jetpack_v2_heartbeat';
+
+	/**
+	 * Singleton
+	 *
+	 * @since 2.3.3
+	 * @static
+	 * @return Heartbeat
+	 */
+	public static function init() {
+		if ( ! self::$instance ) {
+			self::$instance = new Heartbeat();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor for singleton
+	 *
+	 * @since 2.3.3
+	 * @return Heartbeat
+	 */
+	private function __construct() {
+
+		// Schedule the task.
+		add_action( $this->cron_name, array( $this, 'cron_exec' ) );
+
+		if ( ! wp_next_scheduled( $this->cron_name ) ) {
+			// Deal with the old pre-3.0 weekly one.
+			$timestamp = wp_next_scheduled( 'jetpack_heartbeat' );
+			if ( $timestamp ) {
+				wp_unschedule_event( $timestamp, 'jetpack_heartbeat' );
+			}
+
+			wp_schedule_event( time(), 'daily', $this->cron_name );
+		}
+
+		add_filter( 'jetpack_xmlrpc_methods', array( __CLASS__, 'jetpack_xmlrpc_methods' ) );
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			\WP_CLI::add_command( 'jetpack-heartbeat', array( $this, 'cli_callback' ) );
```

Comment:
This makes the command `wp jetpack-heartbeat`, so I guess either the code of the PR description need to change 🙂